### ISSUE TITLE
StatusBar background problem

### ISSUE THREAD

**Issue by ZH3057** [hackiftekhar/IQKeyboardManager#1158]
![2017-11-29 10_43_33](https://user-images.githubusercontent.com/16612221/33355368-325fe1b8-d4f2-11e7-9351-b11b5989bce3.gif)

Using IQKeyboardManager (5.0.6)
iOS 11.1 Simulator iPhone SE 
Xcode Version 9.1
I run the demo on my iPhone SE device, there is the same problom.
I found the same issue #1144 #1145 #1122 #1087 #1189 #1190 #1162

I have runned the latest demo you provide, the same issue below
I test the demo, the controller is presented(presentViewController) cause the issue
![snip20171129_37](https://user-images.githubusercontent.com/16612221/33356610-5819a032-d4f8-11e7-9d99-2b7d6bf63197.png)

@hackiftekhar

**Comment by IPOSDai** [comment]
you should add a scrollView, and all subviews are added on scrollView

**Comment by ZH3057** [comment]
@IPOSDai I think it's unnecessary to do that, do you have another way to fix it.

**Comment by IPOSDai** [comment]
暂且没有，你可以看看  IQKeyboardManager.m 中 adjustFrame方法，就大概明白了

**Comment by KBvsMJ** [comment]
@IPOSDai  @ZH3057  尝试这样修复重写你当前ViewController的loadView方法代码如下:
```

- (void)loadView{
    [super loadView];
    UIScrollView *scrollerView = [[UIScrollView alloc]initWithFrame:[UIScreen mainScreen].bounds];
    scrollerView.contentSize = CGSizeMake(SCREEN_WIDTH, SCREEN_HEIGHT);
    
    self.view = scrollerView;
    
}

```

**Comment by 24KBigDiaoMan** [comment]
@ZH3057   I think it's the offset that causes the problem

 if (@available(iOS 11.0, *)) {
        scrollView.contentInsetAdjustmentBehavior=UIScrollViewContentInsetAdjustmentNever;
        
    }else{
        
        self.automaticallyAdjustsScrollViewInsets=NO;
        
    }

**Comment by mrikh** [comment]
Anyone find a fix for this? Because the entire presented view controller is shifted below the status bar so i don't think it has anything to with the scroll view.....and i'm using the default ios navigation bar and even that doesn't cover the status bar anymore once the keyboard comes up on a presented view controller!

**Comment by deskOfDafa** [comment]
This problem also troubled me for a few days, for my project, is this solution.
Replace the methods in IQKeyboardManager.m

`- (void)keyboardDidHide:(NSNotification*)aNotification
{
    CFTimeInterval startTime = CACurrentMediaTime();
    [self showLog:[NSString stringWithFormat:@"****** %@ started ******",NSStringFromSelector(_cmd)]];
    
    _topViewBeginRect = CGRectZero;
    UIView *statusBar = [[[UIApplication sharedApplication] valueForKey:@"statusBarWindow"] valueForKey:@"statusBar"];
    
    if ([statusBar respondsToSelector:@selector(setBackgroundColor:)]) {
        
        statusBar.backgroundColor = [UIColor whiteColor];
    }
#ifdef __IPHONE_11_0
    if (@available(iOS 11.0, *)) {
        self.initialAdditionalSafeAreaInsets = UIEdgeInsetsZero;
    }
#endif
    
    _kbSize = CGSizeZero;

    CFTimeInterval elapsedTime = CACurrentMediaTime() - startTime;
    [self showLog:[NSString stringWithFormat:@"****** %@ ended: %g seconds ******",NSStringFromSelector(_cmd),elapsedTime]];
}`


A better experience is to change the navigation bar color of the current controller.

`self.navigationController.navigationBar.translucent = NO;
[self.navigationController.navigationBar setBarTintColor:[UIColor whiteColor]];`

**Comment by fbarraganef** [comment]
Any information related to this issue? because I have the same problem but I noticed that this problem happen only with TextView because with TextField everything is working fine for me.

Thanks,
Francisco

**Comment by rdaneshvar95** [comment]
No solution yet? @hackiftekhar

**Comment by adamup928** [comment]
Experiencing the same issue on iOS 11, devices from iPhone 7 to X

**Comment by hackiftekhar** [comment]
The issue is causing because library is trying to adjust the top most controller which is UINavigationController in most of the cases. I did some research about moving to another solution and I found that if library will adjust the currentViewController's frame instead of top most controller's frame then most of the problem will be solved including status bar and navigation bar related issues.

I'm planning to update the library logic to do this change, however let me see how this change will affect other projects. If all goes well, we can do a major release.
